src/main/scala/com/exini/dicom/data/Value.scala

```diff
@@ -653,11 +653,14 @@ object Value {
   def parsePersonName(s: String): Option[PersonName] = {
     def ensureLength(ss: Seq[String], n: Int) = ss ++ Seq.fill(math.max(0, n - ss.length))("")
 
-    val comps = ensureLength(s.split("""\^""").toSeq, 5)
-      .map(s => ensureLength(s.split("=").toSeq, 3).map(trim))
-      .map(c => ComponentGroup(c.head, c(1), c(2)))
+    def transpose(matrix: Seq[Seq[String]]): Seq[Seq[String]] =
+      matrix(0).zipWithIndex.map { case (_, i) => matrix.map(col => col(i)) }
 
-    Option(PersonName(comps.head, comps(1), comps(2), comps(3), comps(4)))
+    val matrix = ensureLength(s.split("=").toSeq, 3)
+      .map(trim)
```

Comment:
This looks wrong to me. `ensureLength` will pad the string with spaces to give you an even length value as required in DICOM - doesn't `trim` remove this space again?

Reply:
Same question for the js library - just realized.